### PR TITLE
[refactor] Fix "const CompileConfig *" to "const CompileConfig &"

### PR DESCRIPTION
**Pull request by PGZXB**
Issue: #7002, https://github.com/taichi-dev/taichi/pull/7159#discussion_r1070803814

### CODE REVIEW

File: taichi/runtime/llvm/llvm_context.cpp [hunk 1]
```diff
@@ -153,7 +153,7 @@ llvm::Type *TaichiLLVMContext::get_data_type(DataType dt) {
     auto num_elements = tensor_type->get_num_elements();
     // Return type is <element_type * num_elements> if real matrix is used,
     // otherwise [element_type * num_elements].
-    if (codegen_vector_type(config_)) {
+    if (codegen_vector_type(*config_)) {
```
Comment (lin-hitonami) [inline taichi/runtime/llvm/llvm_context.cpp:156]: Can we change `config_` to `const CompileConfig &` too? Or will we change/remove it in future PRs?

File: taichi/runtime/program_impls/llvm/llvm_program.cpp [hunk 2]
```diff
@@ -90,12 +90,12 @@ void LlvmProgramImpl::materialize_snode_tree(SNodeTree *tree,
 std::unique_ptr<AotModuleBuilder> LlvmProgramImpl::make_aot_module_builder(
     const DeviceCapabilityConfig &caps) {
   if (config->arch == Arch::x64 || config->arch == Arch::arm64) {
-    return std::make_unique<cpu::AotModuleBuilderImpl>(config, this);
+    return std::make_unique<cpu::AotModuleBuilderImpl>(*config, this);
```
Comment (lin-hitonami) [inline taichi/runtime/program_impls/llvm/llvm_program.cpp:93]: ditto

File: taichi/runtime/llvm/llvm_context.cpp (hunk 1 above)
Comment (PGZXB) [reply taichi/runtime/llvm/llvm_context.cpp:156]: Thanks for your suggestion. I will change it in next PR.

File: taichi/runtime/program_impls/llvm/llvm_program.cpp (hunk 2 above)
Comment (PGZXB) [reply taichi/runtime/program_impls/llvm/llvm_program.cpp:93]: Thanks for your suggestion. I will change it in next PR.

**Review by lin-hitonami — APPROVED** [review]
lgtm

**Pull request merged**